src/cswrap-core.c

```diff
@@ -366,6 +366,66 @@ static bool read_custom_opts(char **dst, const char *str)
     }
 }
 
+#if defined(PATH_TO_CSCLNG) || defined(PATH_TO_CSGCCA) \
+    || defined(PATH_TO_CSMATCH) || defined(PATH_TO_CSCPPC)
+/* if `path` starts with `prefix`, remove the prefix and the trailing ':' */
+static bool remove_path_prefix(char *path, const char *prefix)
+{
+    const size_t prefix_len = strlen(prefix);
+    if (strncmp(path, prefix, prefix_len))
+        // prefix not found
+        return false;
+
+    // skip all ':' that follow the prefix
+    const char *end = path + prefix_len;
+    while (*end == ':')
+        ++end;
+
+    // remove the prefix, including the trailing double dots
+    const size_t new_size = strlen(end) + /* NUL */ 1;
+    memmove(path, end, new_size);
+    return true;
+}
+
+/* if `path` starts with path to a known wrapper, remove the prefix */
+static bool remove_known_wrapper_from_path(char *path)
+{
+    bool removed = false;
+#ifdef PATH_TO_CSCLNG
+    if (remove_path_prefix(path, PATH_TO_CSCLNG))
+        removed = true;
+#endif
+#ifdef PATH_TO_CSGCCA
+    if (remove_path_prefix(path, PATH_TO_CSGCCA))
+        removed = true;
+#endif
+#ifdef PATH_TO_CSMATCH
+    if (remove_path_prefix(path, PATH_TO_CSMATCH))
+        removed = true;
+#endif
+#ifdef PATH_TO_CSCPPC
+    if (remove_path_prefix(path, PATH_TO_CSCPPC))
+        removed = true;
+#endif
+    return removed;
+}
+
+/* remove paths to known wrappers from the beginning of $PATH */
+static void sanitize_path_for_analyzer(void)
+{
+    char *path = getenv("PATH");
+    if (!path || !*path)
+        return;
+
+    while (remove_known_wrapper_from_path(path))
+        ;
+}
+#else
+static void sanitize_path_for_analyzer(void)
+{
+}
```

Comment:
What about this?  It's shorter:
```suggestion
#define sanitize_path_for_analyzer
```

Reply:
Fixed in 1bbbfa0b82e8bcce74e636eb54d9eecaa9bf3b79.